tests/test_rest_api/test_limsdb/test_limsdb.py

```diff
@@ -0,0 +1,481 @@
+mocked_sample.project_name = 'X99999'
+mocked_sample.species = 'Homo sapiens'
+mocked_sample.add_completed_process('Receive Sample EG 6.1', datetime.strptime('01-06-15', '%d-%m-%y'))
+mocked_sample.add_completed_process('Read and Eval SSQC', datetime.strptime('16-07-15', '%d-%m-%y'))
+
+mocked_request = patch('rest_api.limsdb.queries.sample_status.request', return_value=mocked_sample)
+mocked_retrieve_args = patch('rest_api.limsdb.queries.sample_status.retrieve_args', return_value={'match': {'sample_id': 'X99999P001H05', 'project_id': 'X99999'}})
+mocked_get_project_info = patch('rest_api.limsdb.queries.get_project_info', return_value=[('X99999', datetime(2016, 9, 1, 13, 0), 'Jane', 'Doe', 'Number of Quoted Samples', '2')])
+
+mocked_get_sample_info = patch(
+    'rest_api.limsdb.queries.get_sample_info', return_value=[
+        ('X99999', 'X99999P001H05', 'X99999P001', 6, 1, 'Prep Workflow', 'TruSeq PCR-Free DNA Sample Prep'),
+        ('X99999', 'X99999P001H05', 'X99999P001', 6, 1, 'Species', 'Homo sapiens')
+    ]
+)
+
+mocked_get_samples_and_processes = patch(
+    'rest_api.limsdb.queries.get_samples_and_processes',
+    return_value=[
+        ('X99999', 'X99999P001H05', 'Receive Sample EG 6.1', 'COMPLETE', datetime(2016, 11, 11, 11, 45, 3, 367000), 39005),
+        ('X99999', 'X99999P001H05', 'Sequencing Plate Preparation EG 2.0', 'COMPLETE', datetime(2016, 11, 23, 13, 34, 50, 491000), 39917),
+        ('X99999', 'X99999P001H05', 'Create CFP Batch', 'COMPLETE', datetime(2016, 11, 24, 10, 49, 36, 980000), 39934),
+        ('X99999', 'X99999P001H05', 'Read and Eval SSQC', 'COMPLETE', datetime(2016, 11, 25, 13, 56, 13, 175000), 40004)
+    ]
+)
+
+mocked_non_QC_queues = patch(
+    'rest_api.limsdb.queries.non_QC_queues', return_value=[
+        ('X99999', 'X99999P001H05', 'Create PDP Pool', datetime(2016, 11, 27, 11, 14, 59, 325000), 751)
+    ]
+)
+
+
+class SampleTest(TestCase):
+
+    def setUp(self):
+        self.sample1 = Sample()
+        self.sample1.sample_name = 'test_sample'
+        self.sample1.project_name = 'test_project'
+        self.sample1.species = 'Homo sapiens'
+        self.sample1.add_completed_process('Receive Sample EG 6.1', datetime.strptime('01-06-15', '%d-%m-%y'))
+        self.sample1.add_completed_process('Read and Eval SSQC', datetime.strptime('16-07-15', '%d-%m-%y'))
+
+    def tearDown(self):
+        pass
+
+    def test_add_completed_process(self):
+
+        self.sample1.add_completed_process('Test Process', datetime.strptime('01-01-11', '%d-%m-%y'))
+        assert self.sample1._processes == {('Receive Sample EG 6.1', datetime(2015, 6, 1, 0, 0), 'complete', None),
+                                           ('Read and Eval SSQC', datetime(2015, 7, 16, 0, 0), 'complete', None),
+                                           ('Test Process', datetime(2011, 1, 1, 0, 0), 'complete', None)}
+        self.sample1._processes.remove(('Test Process', datetime(2011, 1, 1, 0, 0), 'complete', None))
```

Comment:
You do not need to remove the process as the next `setUp` will overwrite `self.sample1`